packages/docs/docs/video-uploads.mdx

```diff
@@ -135,6 +135,214 @@ As soon as possible, replace the `blob:` URL with the real URL.
 Since `blob:` URLs don't support the HTTP `Range` header, video seeking performance is degraded when using `blob:` URLs.
 :::
 
+## Validating video compatibility
+
+Before uploading a video, you should validate whether the browser can decode it. This saves bandwidth and provides immediate feedback to the user. The recommended best practice is:
+
+1. Check the video locally using a blob URL with Mediabunny's [canDecode()](/docs/mediabunny/can-decode) function
+2. If it can be decoded, upload it to the cloud
+3. If it cannot be decoded, either re-encode it with FFmpeg or reject the upload
+
+### Check if video can be decoded
+
+Use the `canDecode()` snippet to validate the video before uploading:
+
+```ts twoslash title="can-decode.ts"
+import {ALL_FORMATS, Input, UrlSource} from 'mediabunny';
+
+export const canDecode = async (src: string) => {
+  const input = new Input({
+    formats: ALL_FORMATS,
+    source: new UrlSource(src),
+  });
+
+  try {
+    await input.getFormat();
+  } catch {
+    return false;
+  }
+
+  const videoTrack = await input.getPrimaryVideoTrack();
+  if (videoTrack && !(await videoTrack.canDecode())) {
+    return false;
+  }
+
+  const audioTrack = await input.getPrimaryAudioTrack();
+  if (audioTrack && !(await audioTrack.canDecode())) {
+    return false;
+  }
+
+  return true;
+};
+```
+
+### Complete example with validation
+
+This example validates the video locally before uploading, then either uploads it directly or sends it to a re-encoding API endpoint:
+
+```tsx twoslash title="ValidatedUploader.tsx"
+const canDecode = async (src: string) => {
+  return true;
+};
+const upload = async (file: File) => {
+  return 'https://example.com/video.mp4';
+};
+const MyComposition: React.FC<{videoUrl: string}> = ({videoUrl}) => {
+  return null;
+};
+// ---cut---
+import {Player} from '@remotion/player';
+import {useCallback, useState} from 'react';
+
+type UploadState = {type: 'idle'} | {type: 'validating'} | {type: 'uploading'} | {type: 'reencoding'} | {type: 'ready'; url: string} | {type: 'error'; message: string};
+
+export const ValidatedUploader: React.FC = () => {
+  const [state, setState] = useState<UploadState>({type: 'idle'});
+
+  const handleUpload = useCallback(async (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (!file) return;
+
+    try {
+      // Check if video can be decoded using blob URL
+      setState({type: 'validating'});
+      const blobUrl = URL.createObjectURL(file);
+      const isCompatible = await canDecode(blobUrl);
+      URL.revokeObjectURL(blobUrl);
+
+      if (!isCompatible) {
+        // Video needs re-encoding
+        setState({type: 'reencoding'});
+        const formData = new FormData();
+        formData.append('video', file);
+
+        const response = await fetch('/api/reencode-video', {
```

Comment:
In this snippet, we upload the file, re-encode it, download it, and upload it again.

Nobody should implement the flow like this.